develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,384 @@
+    registrations instead.
+
+    Each upgrade is represented by an upgrade registration. An upgrade 
+    registration is an abstraction for the changes you need to apply to the 
+    database from one version to the next one.
+
+    To define a registration, you need to provide
+
+    - the bundle symbolic name of the module.
+    - the schema version your module wants to upgrade from (as a `String`)
+    - the schema version your module wants to upgrade to (as a `String`)
+    - a list of `UpgradeSteps` instances
+
+    For example, here is an upgrade process for the 
+    `com.liferay.calendar.service` module:
+
+        @Override
+    
+        public void register(Registry registry) {
+        
+            registry.register(
+                "com.liferay.calendar.service", "0.0.1", "1.0.0",
+                new com.liferay.calendar.upgrade.
+                v1_0_0.UpgradeCalendarBooking()));    
+        }
+
+    In this example, the `com.liferay.calendar.service` module is being upgraded 
+    from version 0.0.1 to version 1.0.0. The changes are produced by a list of 
+    `UpgradeSteps` instances, which in this example contains only one step:
+
+        new com.liferay.calendar.upgrade.v1_0_0.UpgradeCalendarBooking()
+
+    The internal steps defined within the intermediate classes, the former 
+    `UpgradeProcess` class, require no change on your part, as they are indeed 
+    `UpgradeSteps`. The new framework will process the steps as they are. The
+    example below shows a comparison between the old framework version and the 
+    new framework version.
+
+    old framework:
+
+        public class UpgradeProcess_1_0_1 extends UpgradeProcess {
+        
+            @Override
+            public int getThreshold() {
+                return 102;
+            }
+        
+            @Override
+            protected void doUpgrade() throws Exception {
+                upgrade(UpgradeCalendarBooking.class);
+            }
+            
+        }
+
+    new framework version:
+    
+        @Component(
+            immediate = true,
+            service = {CalendarWebUpgrade.class, UpgradeStepRegistrator.class}
+        )
+        public class CalendarWebUpgrade implements UpgradeStepRegistrator {
+
+            @Override
+            public void register(Registry registry) {
+                registry.register(
+                    "com.liferay.calendar.web", "0.0.0", "1.0.0",
+                    new DummyUpgradeStep());
+                
+                registry.register(
+                    "com.liferay.calendar.web", "0.0.1", "1.0.0",
+                    new UpgradePortletId(), new UpgradePortletPreferences());
+                
+                registry.register(
+                    "com.liferay.calendar.web", "1.0.0", "1.0.1",
+                    new com.liferay.calendar.web.upgrade.v1_0_1.
+                        UpgradePortletPreferences());    
+            }   
+        
+        }
```

Comment:
That UpgradeStep isn't present in the new code. This code was taken directly from the new process for the calendar app. I can add a faux step that shows that instead of the DummyUpgradeStep() though, so there is some connection.